packages/react-dom/src/__tests__/ReactDOMFiber-test.js

```diff
@@ -966,7 +966,7 @@ describe('ReactDOMFiber', () => {
         container,
       );
 
-      simulateMouseMove(null, firstTarget);
+      simulateMouseMove(firstTarget.parentElement, firstTarget);
```

Comment:
Why does this make sense? We haven't "entered" `firstTarget.parentElement` yet so I don't know if it's valid to leave it already.

Reply:
Good point. If you move from an element that is rendered by react to another then it doesn't fire twice i.e. in 

```jsx
<div onMouseEnter={logEventType}>
      <button
        onMouseEnter={logEventType}
        style={{ border: "2px solid black", margin: 50, padding: 50 }}
      >
        mouseEnter fires twice on me sometimes
      </button>
      <button
        onFocus={() => {}}
        onMouseEnter={logEventType}
        style={{ border: "2px solid black", margin: 50, padding: 50 }}
      >
        mouseEnter fires twice on me sometimes
      </button>
    </div>
```
only a single mouseenter is fired. But if you apply

```diff
-<div onMouseEnter={logEventType}>
+<React.Fragment>
```
then you'll observe firing twice. Probably because in

https://github.com/facebook/react/blob/0cd9a6de557f44dcfe79a4f287ca04b05d674bdc/packages/react-dom/src/events/plugins/EnterLeaveEventPlugin.js#L59-L72

`inst` will be `null` since `related` points to an element that does not have a fiber? (since it's the `rootElement` in `ReactDOM.render(<App />, rootElement)`.